src/edit_distance.jl

```diff
@@ -80,20 +83,76 @@ function edit_distance(G₁::AbstractGraph, G₂::AbstractGraph;
             if k < nv(G₁) # there are still vertices to process in G₁?
                 for v in vs
                     λ⁺ = [λ; (k + 1, v)]
-                    enqueue!(OPEN, λ⁺, cost + subst_cost(k + 1, v) + h(λ⁺) - h(λ))
+                    enqueue!(OPEN, λ⁺, cost + edge_cost(G₁, G₂, λ⁺, edge_insert_cost, edge_delete_cost, edge_subst_cost) + subst_cost(k + 1, v) + h(λ⁺) - h(λ))
                 end
                 λ⁺ = [λ; (k + 1, 0)]
-                enqueue!(OPEN, λ⁺, cost + delete_cost(k + 1) + h(λ⁺) - h(λ))
+                enqueue!(OPEN, λ⁺, cost + edge_cost(G₁, G₂, λ⁺, edge_insert_cost, edge_delete_cost, edge_subst_cost) + delete_cost(k + 1) + h(λ⁺) - h(λ))
             else
                 # add remaining vertices of G₂ to the path
-                λ⁺ = [λ; [(0, v) for v in vs]]
-                total_insert_cost = sum(insert_cost, vs)
+                # sequencially adding vertices and hence edges
+                λ⁺ = collect(λ)
+                total_insert_cost = 0
+                for v in vs
+                    total_insert_cost += insert_cost(v)
+                    u_neighbors = is_directed(G₂) ? (inneighbors(G₂, v)..., outneighbors(G₂, v)...) : neighbors(G₂, v)
+                    for u in u_neighbors
+                        for (w, x) in λ⁺
+                            if x == u
+                                total_insert_cost += edge_insert_cost(w, u)
+                            end
+                        end
+                    end
+                    λ⁺ = [λ⁺; (0, v)]
+                end                    
                 enqueue!(OPEN, λ⁺, cost + total_insert_cost + h(λ⁺) - h(λ))
             end
         end
     end
 end
 
+function edge_cost(G₁::AbstractGraph, G₂::AbstractGraph, λ⁺,
+                    edge_insert_cost::Function,
+                    edge_delete_cost::Function,
+                    edge_subst_cost::Function)
+
+    cu = []
```

Comment:
If this is a vector of `Float64`, help the compiler by providing the type:

```
cu = Vector{Float64}()
```

Reply:
No, it contains edges.

Reply:
Then, you could provide the correct type `Vector{Tuple{Int,Int}}()` It should be better than leaving it as a vector of `Any`.